codegen/src/generator/json.rs

```diff
@@ -56,8 +56,14 @@ impl GenerateProtocol for JsonGenerator {
         use serde_json::from_str;".to_string()
     }
 
-    fn generate_struct_attributes(&self) -> String {
-        "#[derive(Debug, Default, Deserialize, Serialize, Clone)]".to_owned()
+    fn generate_struct_attributes(&self, struct_name: &str) -> String {
+        if can_skip_deserializer(struct_name) {
+            return "#[derive(Default, Serialize)]".to_owned();
```

Comment:
We may want to include `Debug` in these for troubleshooting.  Something to keep in mind.